fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1817,28 +1829,51 @@ private static void checkPositive(int value, String desc) {
      * A builder for the indexing policy.
      */
     public static class IndexingPolicy {
-        public static final IndexingPolicy DEFAULT = new IndexingPolicy();
         @Nullable private final String sourceIndex;
         @Nullable private final Object sourceIndexSubspaceKey; // overrides the sourceIndex
         private final boolean forbidRecordScan;
+        private DesiredAction ifDisabled;
+        private DesiredAction ifWriteOnly;
+        private DesiredAction ifMismatchPrev;
+        private DesiredAction ifReadable;
```

Comment:
It would probably be better if these could be made `final` and thus the configuration object could be immutable. I pointed out all of the places that I think would need to change to make that happen.

Also, how good is our test coverage for the various options here? Ideally, for example, it would be good to have a set of unit tests that were parameterized on previous index state and on `DesiredAction` and confirmed that each combination did the right thing. We may already have all of those tests, though

Reply:
Making these fields final might be an issue when allowing `IndexStatePrecondition`, which is not part of this object, to override them. 
(another issue could be when setting, on some flows, `ifWriteOnly` to REBUILD - but that's easier to solve)..

Reply:
Made final.